x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/recovery_list/recovery_table/shards_table.tsx

```diff
@@ -0,0 +1,252 @@
+import { useAppDependencies } from '../../../../index';
+import { formatDate } from '../../../../services/text';
+
+interface Props {
+  shards: SnapshotRecovery['shards'];
+}
+
+export const ShardsTable: React.FunctionComponent<Props> = ({ shards }) => {
+  const {
+    core: { i18n },
+  } = useAppDependencies();
+  const { FormattedMessage } = i18n;
+
+  const columns = [
+    {
+      field: 'id',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.indexColumnTitle', {
+        defaultMessage: 'ID',
+      }),
+      width: '40px',
+      render: (id: SnapshotRecoveryShard['id'], shard: SnapshotRecoveryShard) =>
+        shard.primary ? (
+          <EuiFlexGroup alignItems="center" gutterSize="xs">
+            <EuiFlexItem grow={false}>{id}</EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiToolTip
+                position="right"
+                content={
+                  <FormattedMessage
+                    id="xpack.snapshotRestore.recoveryList.shardTable.primaryTooltipLabel"
+                    defaultMessage="Primary"
+                  />
+                }
+              >
+                <strong>
+                  <FormattedMessage
+                    id="xpack.snapshotRestore.recoveryList.shardTable.primaryAbbreviationText"
+                    defaultMessage="P"
+                  />
+                </strong>
+              </EuiToolTip>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        ) : (
+          id
+        ),
+    },
+    {
+      field: 'stage',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.stageColumnTitle', {
+        defaultMessage: 'Stage',
+      }),
+    },
+    {
+      field: 'startTimeInMillis',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.startTimeColumnTitle', {
+        defaultMessage: 'Start time',
+      }),
+      render: (startTimeInMillis: SnapshotRecoveryShard['startTimeInMillis']) =>
+        startTimeInMillis ? formatDate(startTimeInMillis) : <EuiLoadingSpinner size="m" />,
+    },
+    {
+      field: 'stopTimeInMillis',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.endTimeColumnTitle', {
+        defaultMessage: 'End time',
+      }),
+      render: (stopTimeInMillis: SnapshotRecoveryShard['stopTimeInMillis']) =>
+        stopTimeInMillis ? formatDate(stopTimeInMillis) : <EuiLoadingSpinner size="m" />,
+    },
+    {
+      field: 'totalTimeInMillis',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.durationColumnTitle', {
+        defaultMessage: 'Duration',
+      }),
+      render: (totalTimeInMillis: SnapshotRecoveryShard['totalTimeInMillis']) =>
+        totalTimeInMillis ? (
+          <FormattedMessage
+            id="xpack.snapshotRestore.recoveryList.shardTable.durationValue"
+            defaultMessage="{seconds} {seconds, plural, one {second} other {seconds}}"
```

Comment:
`#` inside plural will return the number. this way it is `1 second` or `x seconds`

```suggestion
            defaultMessage="{seconds, plural, one {# second} other {# seconds}}"
```